src/forward/lib.jl

```diff
@@ -7,12 +7,13 @@ zerolike(x::Tuple) = zerolike.(x)
 end
 
 # TODO figure out why this made a test fail
-zerolike(x::Union{Module,Type}) = false
+zerolike(x::Union{Module,Type}) = x # false -> x
```

Comment:
I don't think the derivative of `x` should ever just be `x`; if you do `foo = getfield(Main, :foo)` then `foo` should have no derivative. So I think it might be better to overload operations on `Module` and `Type` so that they don't throw.

Reply:
Could you give me an idea of what operations we should overload?

Reply:
Whichever operations fail when the derivative is a zero type. Probable `getfield` at least.